src/core/Akka/Dispatch/ChannelSchedulerExtension.cs

```diff
@@ -0,0 +1,389 @@
+﻿using Akka.Actor;
+using Akka.Configuration;
+using Akka.Event;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+
+namespace Akka.Dispatch
+{
+    public sealed class ChannelTaskSchedulerProvider : ExtensionIdProvider<ChannelTaskScheduler>
+    {
+        public override ChannelTaskScheduler CreateExtension(ExtendedActorSystem system)
+        {
+            return new ChannelTaskScheduler(system);
+        }
+    }
+
+    public sealed class ChannelTaskScheduler : IExtension, IDisposable
```

Comment:
Need XML-DOC comments here

Reply:
Most importantly what we need, for future bug fixing and contributors, a summation of how your algorithm works and what role each component plays.

Reply:
done